play_scraper.py

```diff
@@ -1,29 +1,57 @@
-from bs4 import BeautifulSoup
 from urllib.request import urlopen, urlretrieve
+import os
+from bs4 import BeautifulSoup
+
 
 class PlayPageScraper:
-	def __init__(self, location):
-		self._base_url = "https://play.google.com/store/apps/details?id="
-		self._storage_dir = location
-		
-	def get_html(self, app_id, language=None):
-		url = self._base_url + app_id
-		
-		if language is not None:
-			url += "&hl=" + language
-		
-		page = urlopen(url)
-		soup = BeautifulSoup(page, 'html.parser')
-		
-		return soup
-		
-	def get_icon(self, app_id, language=None, directory=""):
-		html = self.get_html(app_id, language)
-		
-		icon = html.find(class_="T75of sHb2Xb")
-		src = icon["src"]
-		
-		location = self._storage_dir + directory
-
-		_, _ = urlretrieve(src, location + "/icon_" + app_id)
+    """A class for scraping Google Play Store web pages.
+
+    Attributes:
+        _base_url: str
+            Base url of the app's web pages.
+        _storage_dir: str
+            The main storage directory for retrieved info.
+    """
+
+    ICON_CLASS = "T75of sHb2Xb"
+
+    def __init__(self, base_url, storage_dir):
+        self._base_url = base_url
+
+        if not os.path.exists(storage_dir):
+            os.mkdir(storage_dir)
+        self._storage_dir = storage_dir
+
+    def __get_html(self, app_id, language=None):
+        """Obtains the html of the app's web page.
+
+        :param app_id (str): the id of the app.
+        :param language (str): app's language.
+        :return: a BeautifulSoup object containing the html.
+        """
+        url = self._base_url + app_id
+
+        if language is not None:
+            url += "&hl=" + language
+
+        page = urlopen(url)
+        soup = BeautifulSoup(page, 'html.parser')
+        return soup
+
+    def get_icon(self, app_id, language=None, directory=""):
+        """Downloads the app's icon from the corresponding web page.
+
+        :param app_id (str): the id of the app.
+        :param language (str): app's language.
+        :param directory (str): icon storage subdirectory.
+        """
+        html = self.__get_html(app_id, language)
+
+        icon = html.find(class_=self.ICON_CLASS)
+        src = icon["src"]
+
+        location = self._storage_dir + directory
+        if directory and not os.path.exists(location):
+            os.mkdir(location)
 
+        _, _ = urlretrieve(src, location + "icon_" + app_id)
```

Comment:
Using `+` to append a subdirectory to a path is brittle and will silently break if someone does not add a trailing underscore to the parent directory. It is also not platform-independent.
For these reasons I would always use the helpers in `pathlib` or `os.path.join`.
In the suggestion I assume that we changed `storage_dir` to be a `Path` instance.
Note that `Path('a/b/c') / "" == Path('a/b/c') == Path('a/b') / 'c'`, meaning you don't need to handle the case where no `directory` was passed separately.

```suggestion
        location = self._storage_dir / directory
        location.mkdir(exist_ok=True)
        urlretrieve(src, location / f"icon_{app_id}")
```